src/Shared/FileMatcher.cs

```diff
@@ -372,7 +478,7 @@ GetFileSystemEntries getFileSystemEntries
                     else
                     {
                         // getFileSystemEntries(...) returns an empty array if longPath doesn't exist.
-                        string[] entries = getFileSystemEntries(FileSystemEntity.FilesAndDirectories, longPath, parts[i], null, false);
+                        string[] entries = getFileSystemEntries(FileSystemEntity.FilesAndDirectories, longPath, parts[i], null, false).ToArray();
```

Comment:
Minor: ToList() is usually slightly faster, (they are almost same to copy data to a buffer, but ToArray() often copies one more time to fit the exact size.

Reply:
With the [recent change](https://github.com/Microsoft/msbuild/pull/2572/commits/1322acf14ed4e26a88bd3bcc927ea13b94426dd0), the `ToArray` is not needed anymore.